frontend/catalyst/pennylane_extensions.py

```diff
@@ -261,31 +306,53 @@ def circuit(x):
     >>> workflow(2.0)
     array(-3.14159265)
     """
-    methods = {"fd", "ps", "adj"}
-    if method is None:
-        method = "fd"
-    if method not in methods:
-        raise ValueError(
-            f"Invalid differentiation method '{method}'. "
-            f"Supported methods are: {' '.join(sorted(methods))}"
-        )
-    if method == "fd" and h is None:
-        h = 1e-7
-    if not (h is None or isinstance(h, numbers.Number)):
-        raise ValueError(f"Invalid h value ({h}). None or number was excpected.")
-    if argnum is None:
-        argnum = [0]
-    elif isinstance(argnum, int):
-        argnum = [argnum]
+    return Grad(_bless_differentiable(f), grad_params=_check_grad_params(method, h, argnum))
+
+
+def jvp(f, params, tangents, *, method=None, h=None, argnum=None):
+    """
+    Args:
+        f(DifferentiableLike): Function-like object to calculate JVP for
+    """
+
+    def _check(x, hint):
+        # pylint: disable=no-else-return
+        if isinstance(x, list):
+            return x
+        elif isinstance(x, tuple):
+            return list(x)
+        else:
+            raise ValueError(f"jvp '{hint}' argument must be a list or a tuple, not {type(x)}")
+
+    params = _check(params, "params")
+    tangents = _check(tangents, "tangents")
+    fn: Differentiable = _bless_differentiable(f)
+    grad_params = _check_grad_params(method, h, argnum)
+    jaxpr = _make_jaxpr_differentiable(fn, grad_params, *params)
+    return jprim.jvp_p.bind(*(params + tangents), jaxpr=jaxpr, fn=fn, grad_params=grad_params)
 
-    if isinstance(f, catalyst.compilation_pipelines.QJIT):
-        # Don't generate an extra function when the circuit is already qjitted.
-        f = f.qfunc
 
-    if isinstance(f, qml.QNode):
-        return Grad(f, method=method, h=h, argnum=argnum)
+def vjp(f, params, cotangents, *, method=None, h=None, argnum=None):
+    """
+    Args:
+        f(DifferentiableLike): Function-like object to calculate VJP for
```

Comment:
@grwlf don't forget to fill in the details of all the args, the returns, and to also include a minimal example :) See `grad` above for reference